### PR TITLE
fix: preserve updated payment amount

### PR DESCRIPTION
**Pull request by swapnil-turing**
## Summary
- keep existing-payment fields in sync before Stripe session creation
- prevent stale in-memory payment method from being persisted during follow-up save
- extend regression coverage for repayment flow to assert updated payment method

## Notes
- scope is checkout payment serializer behavior and checkout API tests only

### CODE REVIEW

File: augment-store/server/checkout/serializers.py [hunk 1]
```diff
@@ -266,6 +266,7 @@ def create(self, validated_data):
         try:
             payment = order.payment
             Payment.objects.filter(id=payment.id).update(amount=amount, payment_method=payment_method)
+            payment.refresh_from_db(fields=["amount"])
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/serializers.py]: value:useful; category:bug; feedback:Chose useful because this comment caught a state-consistency flaw between in-memory and persisted payment fields during a critical checkout path. It identified that a stale payment method could be written back after session creation, causing stored payment data to diverge from API behavior.

File: augment-store/server/checkout/tests.py [hunk 2]
```diff
@@ -782,6 +788,8 @@ def test_payment_order(self, mock_create_payment_session):
 
         # THEN we should get a 201 response
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        order.payment.refresh_from_db()
+        self.assertEqual(order.payment.amount, order.total)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/tests.py:792]: value:good-to-have; category:bug; feedback:Marked good-to-have because it strengthened regression coverage around a field that can silently drift in edge cases. It highlighted that payment method transitions could regress unnoticed even when amount assertions still pass.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/checkout/serializers.py [hunk 3]
```diff
@@ -266,6 +266,7 @@ def create(self, validated_data):
         try:
             payment = order.payment
             Payment.objects.filter(id=payment.id).update(amount=amount, payment_method=payment_method)
+            payment.refresh_from_db(fields=["amount", "payment_method"])
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/serializers.py:269]: value:good-to-have; category:documentation; feedback:Chose good-to-have because this comment improved reviewer clarity rather than exposing a direct runtime defect. It identified that mismatched PR metadata can mislead review focus and release notes, which increases coordination risk during merge and audit.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/checkout/tests.py [hunk 4]
```diff
@@ -770,6 +770,12 @@ def test_payment_order(self, mock_create_payment_session):
         order = OrderFactory(created_by=self.user)
         cart_item = CartItemFactory(quantity=1)
         OrderItemFactory(order=order, cart_item=cart_item)
+        PaymentFactory(
+            order=order,
+            created_by=self.user,
+            amount=Decimal("1.00"),
+            payment_method=Payment.PaymentMethod.CASH_ON_DELIVERY,
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/checkout/tests.py]: value:useful; category:bug; feedback:Chose useful because this comment caught a concrete test break that would fail before exercising the repayment logic. It identified an invalid enum member usage that raises AttributeError, which would hide real behavior regressions behind a setup error.

**Comment by swapnil-turing** [comment]
augment review

**Review by michaelos443 — APPROVED** [review]
